### PR TITLE
samples: use evutil_socket_t instead  and handle 64 bit Windows

### LINKED ISSUE #1574: HTTP Server sample fails to compile for Windows x64
`evutil_socket_t` is a `long long` on Windows.
The `do_term` function passed as a pointer then does not match the function pointer type declaration. in `http-server.c`

### Expected
libevent builds

### Actual
```
../libevent-2.1.12-stable/sample/http-server.c:559:9: error: incompatible function pointer types passing
      'void (int, short, void *)' to parameter of type 'event_callback_fn' (aka 'void (*)(long long, short, void *)')
      [-Wincompatible-function-pointer-types]
  559 |         term = evsignal_new(base, SIGINT, do_term, base);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../libevent-2.1.12-stable/include/event2/event.h:983:44: note: expanded from macro 'evsignal_new'
  983 |         event_new((b), (x), EV_SIGNAL|EV_PERSIST, (cb), (arg))
      |                                                   ^~~~
../libevent-2.1.12-stable/include/event2/event.h:1090:87: note: passing argument to parameter here
 1090 | struct event *event_new(struct event_base *, evutil_socket_t, short, event_callback_fn, void *);
      |                                                                                       ^
```

### PR DESCRIPTION
**Pull request by Coeur**
Fixes #1574.
Supersedes #1575.
The solution used is identical to the one in becat.c.

### CODE REVIEW

**Review by azat — COMMENTED** [review]
OK. BTW signals are not supported on windows, so let's just hide this code under `#ifndef _WIN32`?

**Comment by Coeur** [comment]
But which part of the code (which line numbers) ?
And if the file is currently compiling fine, why adding macros that would complexify the file?

**Comment by azat** [comment]
> And if the file is currently compiling fine, why adding macros that would complexify the file?

To avoid readers of this file think that signals is supported on windows

**Comment by azat** [comment]
Hm, actually, [windows does support signals](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/signal?view=msvc-170), but they are not asynchronous, plus SIGINT creates a thread - very odd

**Comment by azat** [comment]
It is ok to leave as-is, but the codes does not compile. Please fix

```
http-server.obj : error LNK2019: unresolved external symbol strsignal referenced in function do_term [D:\a\libevent\libevent\build\http-server.vcxproj]
D:\a\libevent\libevent\build\bin\Debug\http-server.exe : fatal error LNK1120: 1 unresolved externals [D:\a\libevent\libevent\build\http-server.vcxproj]
```

**Review by azat — CHANGES_REQUESTED** [review]
.

File: util-internal.h [hunk 1]
```diff
@@ -520,6 +520,10 @@ HMODULE evutil_load_windows_system_library_(const TCHAR *library_name);
 #endif
 #endif
 
+#if !defined(EVENT__HAVE_STRSIGNAL) && !defined(strsignal)
+#define strsignal(sig) "Signal"
```
Comment (azat) [inline util-internal.h:524]: Inline function is prefered over macroses

**Review by azat — APPROVED** [review]

**Pull request merged**